### DISCUSSION TITLE
Can I run Speechly on a HTML page on my local machine?

### DISCUSSION THREAD

**Discussion by arzga** [speechly/speechly#118] [Q&A] [answered]
Can I create a local HTML page with Speechly on it and open it in my web browser? For example, open an `index.html` file with the following content:

```
<html>
  <head>
    <script type="text/javascript" src="https://unpkg.com/@speechly/browser-ui/core/push-to-talk-button.js"></script>
    <script type="text/javascript" src="https://unpkg.com/@speechly/browser-ui/core/big-transcript.js"></script>
    <script type="text/javascript" src="https://unpkg.com/@speechly/browser-ui/core/error-panel.js"></script>
  </head>
  <body>
    <big-transcript placement="top"></big-transcript>
    <push-to-talk-button appid="YOUR_APP_ID_FROM_SPEECHLY_DASHBOARD" placement="bottom"></push-to-talk-button>
    <error-panel placement="bottom"></error-panel>
  </body>
</html>
```

(echoing a frequently asked question from developers)

**Comment by aukkonen** [comment, marked as answer]
It is unfortunately not possible to use Speechly simply by opening local files in your browser. This is due to security restrictions imposed by web browsers when using file:// URLs. For example, access to microphone is disabled in such cases.

The solution is to set up a local web server on your computer, and open the page using the http protocol as you do with remotely hosted websites.

This is not as hard as it may sound. Two simple alternatives that we recommend are to use either Node.js or Python. You can find good instructions for both in [this blog post by Tim Wells](https://timnwells.medium.com/quick-and-easy-http-servers-for-local-development-7a7df5ac25ff).
